src/vm/gdbjit.cpp

```diff
@@ -600,6 +673,157 @@ struct __attribute__((packed)) DebugInfoVar
     uint32_t m_var_type;
 };
 
+struct __attribute__((packed)) DebugInfoClassType
+{
+    uint8_t m_type_abbrev;
+    uint32_t m_type_name;
+    uint8_t m_byte_size;
+};
+
+struct __attribute__((packed)) DebugInfoClassMember
+{
+    uint8_t m_member_abbrev;
+    uint32_t m_member_name;
+    uint32_t m_member_type;
+    uint8_t m_member_loc;
+};
+
+struct __attribute__((packed)) DebugInfoRefType
+{
+    uint8_t m_type_abbrev;
+    uint32_t m_ref_type;
+};
+
+void TypeInfoBase::DumpStrings(char* ptr, int& offset)
+{
+    if (ptr != nullptr)
+    {
+        strcpy(ptr + offset, m_type_name);
+        m_type_name_offset = offset;
+    }
+    offset += strlen(m_type_name) + 1;
+}
+
+void BaseTypeInfo::DumpDebugInfo(char* ptr, int& offset)
+{
+    if (m_type_offset != 0)
+    {
+        return;
+    }
+
+    if (ptr != nullptr)
+    {
+        NewHolder<DebugInfoType> bufType = new (nothrow) DebugInfoType;
```

Comment:
It seems the bufType can be a stack variable, it is small and you copy its contents out after you build it anyways.
And the same at several places below where you use the holder for other debug info stuff.